metadata-db-app/src/cmr/metadata_db/services/generic_documents.clj

```diff
@@ -27,27 +22,21 @@
                   :user-id])
 
 ;; These are fields that should not be inclucded in the :Metadata field because
-;; they are generated by software. Add items to this list corrisponding to 
+;; they are generated by software. Add items to this list corrisponding to
 ;; database column names which should NOT be shared with the world and are not
 ;; in the info-fields list above.
-(def fields-to-drop (into [] (concat info-fields [:id :deleted :transaction-id])))
+(def fields-to-drop (vec (concat info-fields [:id :deleted :transaction-id])))
 
 (defn- raw-generic->response
   "Take the raw database object returned from either SQL or In-Memory and shape
    it into the expected response by pulling out generated fields like concept-id
    and putting them in the :Info field, then putting the metadata inside the
    :Metadata field without any of these extra fields."
   [raw-doc]
-  {:Info (assoc-these-from info-fields raw-doc)
+  {:Info (select-keys raw-doc info-fields)
    :Metadata (reduce #(dissoc %1 %2) raw-doc fields-to-drop)})
```

Comment:
Why are you not getting metadata directly like (select-keys raw-doc [:metadata]), instead of removing all the rest of the unwanted fields?

Reply:
because it is not inside a metadata block, it was all mixed in with all the other fields, I don't know which ones are metadata at this point, but I do know which ones are not. Then, as you can see, I put them all under a map so that nothing outside of metadata-db will know what was done.

Reply:
I would have to go back and clean up the metadata code, but we mixed everything together to simplify our usage of the in memory database.